kafka-rest/src/main/java/io/confluent/kafkarest/resources/v3/ProduceAction.java

```diff
@@ -94,14 +96,16 @@ public ProduceAction(
       Provider<ProducerMetrics> producerMetrics,
       ChunkedOutputFactory chunkedOutputFactory,
       StreamingResponseFactory streamingResponseFactory,
-      RateLimiter rateLimiter) {
+      RateLimiter rateLimiter,
+      ExecutorService executorService) {
     this.schemaManagerProvider = requireNonNull(schemaManagerProvider);
     this.recordSerializerProvider = requireNonNull(recordSerializer);
     this.produceControllerProvider = requireNonNull(produceControllerProvider);
     this.producerMetrics = requireNonNull(producerMetrics);
     this.chunkedOutputFactory = requireNonNull(chunkedOutputFactory);
     this.streamingResponseFactory = requireNonNull(streamingResponseFactory);
     this.rateLimiter = requireNonNull(rateLimiter);
+    this.executorService = requireNonNull(executorService);
```

Comment:
HK2 doesn't look good in our profiles, so I am starting to be much more skeptical to its usage. Unlike most of the other things we inject (like controllers or configs), this also is not really gaining much from being injectable, as it has much stronger locality.

If we have to configure the pool (although we can do without that - see my pool size comments below) we have to consume the injectable config, so I am not recommending to put this as a `static` field, but I will think about a better recommendation.

Reply:
I'm going to leave this as an HK2 injected field - it's just one of many instances of this pattern.  If we determine it's something we want to alter we should come back to it and sort them all, not just this one.